src/quantuminspire/qiskit/qi_job.py

```diff
@@ -77,32 +77,60 @@ def submit(self) -> None:
             raise JobError('Job has already been submitted!')
         self._job_id = self._backend.run(self._qobj)
 
-    def result(self, timeout: Optional[float] = None, wait: float = 0.5, only_latest_run: bool = True) -> QIResult:
-        """
+    def _wait_for_job_finished(self, timeout: Optional[float] = None, wait: float = 0.5) -> None:
+        """ Wait for the job to be finished.
 
         Args:
             timeout: Timeout in seconds.
             wait: Wait time between queries to the quantum-inspire platform.
-            only_latest_run: when True, only the result for the experiments in the latest run for this project are
-            fetched, when False all the experiment results for all the jobs in the project are fetched
-
-        Returns:
-            QIResult object containing results from the experiments.
 
         Raises:
             JobTimeoutError: If timeout is reached.
-            QisKitBackendError: If an error occurs during simulation.
         """
         start_time = time.time()
         while self.status() not in JOB_FINAL_STATES:
             elapsed_time = time.time() - start_time
             if timeout is not None and elapsed_time > timeout:
                 raise JobTimeoutError('Failed getting result: timeout reached.')
             time.sleep(wait)
-        if only_latest_run:
-            experiment_results = self._backend.get_experiment_results_from_latest_run(self)
-        else:
-            experiment_results = self._backend.get_experiment_results_from_all_jobs(self)
+
+    def result(self, timeout: Optional[float] = None, wait: float = 0.5) -> QIResult:
```

Comment:
This can be refactored further:
```
    def _result(self, result_function: Callable[[], List[ExperimentResult]], timeout: Optional[float] = None,
                wait: float = 0.5) -> QIResult:
        self._wait_for_job_finished(timeout, wait)
        experiment_results = result_function()
        return QIResult(backend_name=self._backend.backend_name, backend_version=quantum_inspire_version,
                        job_id=self.job_id(), qobj_id=self.job_id(), success=True, results=experiment_results)

    def result(self, timeout: Optional[float] = None, wait: float = 0.5) -> QIResult:
        return self._result(self._backend.get_experiment_results_from_latest_run, timeout, wait)

    def result_all_jobs(self, timeout: Optional[float] = None, wait: float = 0.5) -> QIResult:
        return self._result(self._backend.get_experiment_results_from_all_jobs, timeout, wait)


```

Reply:
I don't think shorter is always better. The 'function pointers' gives additional complexity in the resulting code. But I'll change again. I undo the _wait_for_job_finished extract method refactoring.